lib/factory_girl/registry.rb

```diff
@@ -13,6 +13,10 @@ def clear
       @items.clear
     end
 
+    def remove(name)
+      @items.delete(name.to_sym)
```

Comment:
I would not want to delete something unless I knew it's causing problems, but if a future release of that dependency removes the factory altogether, I don't know if I really want it raising -- the same side effects are there either way, there is no factory present with that name after the line of code is executed. 

But I see what you're saying, if you feel strongly about it I can make it raise. Which would eliminate the other question over what it should return, probably.